### PR TITLE
feat(cli): add generic custom flag builders for command-local flags

### PR DESCRIPTION
**Pull request by galligan**
## Summary

- Adds four typed flag builders: `booleanFlagPreset`, `enumFlagPreset`, `numberFlagPreset`, `stringListFlagPreset`
- Each builder follows the same pattern: config → `createPreset` → typed resolve with source key fallbacks
- Includes input guards on `numberFlagPreset` to reject empty string, null, and boolean coercion
- Dogfoods builders immediately on `outfitter upgrade` action (replaces `resolveNoCodemodsFlag`)

## Test plan

- [x] All four builder types tested with happy paths and edge cases
- [x] `numberFlagPreset` guards against `Number("")`, `Number(null)`, `Number(true)` coercion
- [x] `integer: false` fractional value support verified
- [x] Composition with `composePresets` works correctly

Closes OS-220

### CODE REVIEW

File: packages/cli/src/flags.ts [hunk 1]
```diff
@@ -146,6 +150,202 @@ export function composePresets<
...
+  const sources = resolveSourceKeys(config.key, config.sources);
+  const defaultValue = config.defaultValue ?? false;
+
+  return createPreset({
+    id: config.id,
+    options: [
+      {
+        flags: config.flags,
+        description: config.description,
+        defaultValue,
```
[the remark this answers is not in the document]
Comment (galligan) [reply packages/cli/src/flags.ts]: Fixed — `booleanFlagPreset` now detects `--no-*` flags and omits `defaultValue` from the Commander option definition, preserving Commander's natural default (`true` for the positive attribute). Added test asserting the option has no default for negated flags.
